lib/cli/cli.js

```diff
@@ -45,10 +44,13 @@ exports.main = (argv = process.argv.slice(2)) => {
       'Options:': 'Other Options',
       'Commands:': 'Commands'
     })
-    .fail((msg, err, yargs) => {
+    .fail((msg, err) => {
       debug(err);
-      yargs.showHelp();
-      console.error(`\n${symbols.error} ${ansi.red('ERROR:')} ${msg}`);
+      if (err) {
+        console.error(`\n${err.stack || err}`);
+      } else {
+        console.error(`\nError: ${msg}`);
+      }
```

Comment:
By this change the error formatting is lost,
eg. see "test/integration/fixtures/options/forbid-only/only-suite.fixture.js".
IMO it's better to leave this code untouched and to handle errors earlier in the yargs handler, also to avoid `UnhandledPromiseRejectionWarning`.

Reply:
I did it, and reverted it back to what it was! As I said above, the weird running "sync & async into "always async" forced me to do it this way, and moving to "always async" fixed it and allowed me to revert to the correct way of doing this.